src/main/java/com/aws/greengrass/util/platforms/windows/WindowsExec.java

```diff
@@ -5,29 +5,46 @@
 
 package com.aws.greengrass.util.platforms.windows;
 
+import com.aws.greengrass.config.PlatformResolver;
+import com.aws.greengrass.lifecyclemanager.KernelAlternatives;
 import com.aws.greengrass.util.Exec;
 import com.aws.greengrass.util.Utils;
+import com.aws.greengrass.util.platforms.Platform;
+import com.aws.greengrass.util.platforms.UserPlatform;
 import org.zeroturnaround.process.Processes;
+import vendored.com.microsoft.alm.storage.windows.internal.WindowsCredUtils;
 
 import java.io.File;
 import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.CharBuffer;
+import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 import java.util.concurrent.TimeUnit;
 import javax.annotation.Nullable;
+import javax.inject.Inject;
+
+import static com.aws.greengrass.config.PlatformResolver.ARCHITECTURE_KEY;
+import static com.aws.greengrass.config.PlatformResolver.ARCH_X86;
 
 @SuppressWarnings("PMD.AvoidCatchingThrowable")
 public class WindowsExec extends Exec {
-    public static final String PATHEXT_KEY = "PATHEXT";
-
-    private static final List<String> PATHEXT;  // ordered file extensions to try, when no extension is provided
+    private final List<String> pathext;  // ordered file extensions to try, when no extension is provided
+    private final String runasExePath;
 
-    static {
-        String pathExt = System.getenv(PATHEXT_KEY);
-        PATHEXT = Arrays.asList(pathExt.split(File.pathSeparator));
+    @Inject
+    WindowsExec(PlatformResolver platformResolver, KernelAlternatives kernelAlts) {
+        super();
+        String pathExt = System.getenv("PATHEXT");
+        pathext = Arrays.asList(pathExt.split(File.pathSeparator));
+        Path runasPath = ARCH_X86.equals(platformResolver.getCurrentPlatform().get(ARCHITECTURE_KEY))
```

Comment:
are we getting CPU arch correctly for windows? On linux we use uname

Reply:
This one we're using `System.getProperty("os.arch")`. There is also `ARCHITECTURE_DETAIL_KEY` which we call uname and do nothing on windows

Reply:
Sounds like that needs a fix then

Reply:
Looks like archDetail is for lambda to know which exact ARM arch https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/main/src/main/java/com/aws/greengrass/config/PlatformResolver.java#L141
We're not worrying about lambda right now